src/common/catalog/catalog-category-registry.ts

```diff
@@ -25,32 +25,30 @@ import { CatalogCategory, CatalogEntityData, CatalogEntityKindData } from "./cat
 
 export class CatalogCategoryRegistry {
   protected categories = observable.set<CatalogCategory>();
+  protected groupKinds = new ExtendedMap<string, ExtendedMap<string, CatalogCategory>>();
 
   @action add(category: CatalogCategory): Disposer {
     this.categories.add(category);
+    this.updateGroupKinds(category);
 
-    return () => this.categories.delete(category);
+    return () => {
+      this.categories.delete(category);
+      this.groupKinds.clear();
```

Comment:
This is wrong FYI @jakolehm  We don't want to clear the whole map.

Reply:
The reason that I used `@computed` is because it is sort of like a computed index in a database table.

This should be:
```suggestion
      this.groupKinds.get(category.spec.group).delete(category.spec.names.kind);
```